lib/rubocop/cop/style/mixin_usage.rb

```diff
@@ -48,24 +48,22 @@ class MixinUsage < Cop
 
         def_node_matcher :include_statement, <<-PATTERN
           (send nil? ${:include :extend :prepend}
-            (const nil? _))
+            const)
         PATTERN
 
         def on_send(node)
           return unless (statement = include_statement(node))
-          return unless top_level_node?(node)
+          return unless implicit_style?(node)
 
           add_offense(node, message: format(MSG, statement: statement))
         end
 
         private
 
-        def top_level_node?(node)
-          if node.parent.parent.nil?
-            node.sibling_index.zero?
-          else
-            top_level_node?(node.parent)
-          end
+        def implicit_style?(node)
```

Comment:
I think it should be recursive, and check class/module types.
For example: (but the example does not pass `Class.new` test case.)

```ruby
def implicit_style?(node)
  parent = node.parent
  return true unless parent
  return false if parent.respond_to?(:macro?) && parent.macro?
  return false if parent.class_type? || parent.module_type?
  implicit_style?(parent)
end
```



Currently the cop does not add offence for the following code.


```ruby
some_method do
  include A
end
```

Reply:
How about something like this?

```ruby
def accepted_include?(node)
  return true unless node.macro?
 
  node.each_ancestor(:class, :module, :send, :block).any? do |ancestor|
    accepted_include_context?(ancestor)
  end
end

def accepted_include_context?(ancestor)
  ancestor.class_type? || ancestor.module_type? || dynamic_class_definition?(ancestor)
end

def dynamic_class_definition?(ancestor)
  # Check for Class.new, class_eval, etc.
end
```

(I'm not sure

Reply:
I don't understand 2 things 😓 

1. Test cases that need recursive and class/module types
2. About the following use case

```ruby
some_method do
  include A
end
```

## 1. Test cases that need recursive and class/module types

Even without recursive and class/module types, there is no problem with existing bug cases (The test cases that currently exists passes). I could not make a test cases that fails without recursive and class/module types.

## 2. About the following use case

```ruby
some_method do
  include A
end
```

For this use case, I'd like to support with another PR at another opportunity. First of all, in this PR, I'd like to solve the known issue of informing what is not a offense as a offense.